coredb-operator/src/controller.rs

```diff
@@ -139,6 +147,50 @@ impl CoreDB {
             .map_err(Error::KubeError)?;
         Ok(Action::await_change())
     }
+
+    async fn primary_pod(&self, client: Client) -> Result<Pod> {
+        let sts = stateful_set_from_cdb(self);
+        let sts_name = sts.metadata.name.unwrap();
+        let sts_namespace = sts.metadata.namespace.unwrap();
+        let label_selector = format!("statefulset={}", sts_name);
+        let list_params = ListParams::default().labels(&label_selector);
+        let pods: Api<Pod> = Api::namespaced(client, &sts_namespace);
+        let pods = pods.list(&list_params);
+        // For the time being, we assume that the first pod is the primary
```

Comment:
At the moment, we are just running one replica to avoid needing to handle service routing to the primary. We will need some kind of logic to pick the primary pod and point the service at it once we are doing standby / failover pods.